pkg/pac/pac.go

```diff
@@ -218,6 +206,72 @@ func addWebhook(client *gitlab.Client, projectID int, webhookURL, token string)
 	return nil
 }
 
+// Create a new Repository under current namespace
+func createNewRepository(c *clients.Clients, projectName, targetGroupNamespace, namespace string) error {
+	repo := &pacv1alpha1.Repository{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "pipelinesascode.tekton.dev/v1alpha1",
+			Kind:       "Repository",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      projectName,
+			Namespace: namespace,
+		},
+		Spec: pacv1alpha1.RepositorySpec{
+			URL: fmt.Sprintf("https://gitlab.com/%s/%s", targetGroupNamespace, projectName),
+			GitProvider: &pacv1alpha1.GitProvider{
+				URL: "https://gitlab.com",
+				Secret: &pacv1alpha1.Secret{
+					Name: webhookConfigName,
+					Key:  "provider.token",
+				},
+				WebhookSecret: &pacv1alpha1.Secret{
+					Name: webhookConfigName,
+					Key:  "webhook.secret",
+				},
+			},
+		},
+	}
+
+	repo, err := c.PacClientset.Repositories(namespace).Create(context.Background(), repo, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create repository: %v", err)
+	}
+
+	log.Printf("Repository '%s' created successfully in namespace '%s'", repo.GetName(), repo.GetNamespace())
+	return nil
+}
+
+func SetupGitLabProject(client *gitlab.Client) *gitlab.Project {
+	gitlabGroupNamespace := os.Getenv("GITLAB_GROUP_NAMESPACE")
+	projectIDOrPath := os.Getenv("GITLAB_PROJECT_ID")
+
+	if gitlabGroupNamespace == "" || projectIDOrPath == "" {
+		testsuit.T.Fail(fmt.Errorf("failed to get system variables"))
+	}
+
+	smeeURL := store.GetScenarioData("SMEE_URL")
+	token := config.TriggersSecretToken
```

Comment:
My run failed with this error

```
{"level":"error","ts":"2024-10-25T12:36:40.862Z","logger":"pipelinesascode","caller":"events/emit.go:46","msg":"cannot create status: could not validate payload, check your webhook secret?: gitlab failed validation: event's secret doesn't match with webhook secret: no gitlab client has been initialized, exiting... (hint: did you forget setting a secret on your repo?)","provider":"gitlab","event-id":"","event-sha":"32b7eb894ecaf6bff9ba4792be24a507175edf43","event-type":"Merge Request","namespace":"releasetest-ndxtg","stacktrace":"github.com/openshift-pipelines/pipelines-as-code/pkg/events.(*EventEmitter).EmitMessage\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/events/emit.go:46\ngithub.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode.(*PacRun).Run\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode/pipelineascode.go:67\ngithub.com/openshift-pipelines/pipelines-as-code/pkg/adapter.(*sinker).processEvent\n\t/go/src/github.com/openshift-pipelines/pipeli...
```

IMHO you should read the value of env variable `WEBHOOK_TOKEN` here

Reply:
Fixed it. Thanks for pointing it out.